src/relations/db.py

```diff
@@ -159,15 +157,9 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
         # is connecting to this database will receive a "database gone" event from the
         # old PostgreSQL library (ops-lib-pgsql) and the connection between the
         # application and this charm will not work.
-        allowed_subnets = self._get_allowed_subnets(event.relation)
-        allowed_units = self._get_allowed_units(event.relation)
-        for databag in [application_relation_databag, unit_relation_databag]:
+        for databag in [unit_relation_databag]:
```

Comment:
I don't think we need to loop here any more.

Reply:
Good catch. I forgot to remove it. I'm going to fix it.